File: N-th Fibonacci Number.py
from os import *

from sys import *

from collections import *

from math import *

 

def fibonacciNumber(n):

    a = 0

    b = 1

    

    if n == 0:

        return a

    elif n == 1:

        return b

    else:

        for i in range(n-1):

            c = (a + b) % (10 ** 9 + 7)

            a = b

            b = c

        return c

File: N-th Fibonacci Number.java
import java.util.* ;
import java.io.*; 
public class Solution{
    public static int fibonacciNumber(int n){
        // Write your code here.
         int n1 = 0;

        int n2 = 1;

        int result=0;

        int mod = ((int)Math.pow(10,9))+7;

        for(int i=2; i<=n; i++){

            result =  n1+n2;

            n1=n2;

            n2=result;

            if(n2>+mod){

                n2 -=mod;

            }

        }

        return n2;
    }
}
